### PR TITLE
Add InitOptions to configure Validator

### PR DESCRIPTION
**Pull request by hkundag**
InitOptions are passed to NewValidator along with paths to customize the
Validator configuration. So far, it only supports DisabledBuiltins,
which allows users to disable certain functions in Rego.

### CODE REVIEW

File: pkg/gcv/validator.go [hunk 1]
```diff
@@ -86,9 +90,14 @@ func NewValidatorConfig(policyPaths []string, policyLibraryPath string) (*config
 func newCFClient(
 	targetHandler cfclient.TargetHandler,
 	templates []*cftemplates.ConstraintTemplate,
-	constraints []*unstructured.Unstructured) (
+	constraints []*unstructured.Unstructured,
+	options *InitOptions) (
```
Comment (briantkennedy) [inline pkg/gcv/validator.go:94]: structure this as a standard Go option

Comment (hkundag) [reply pkg/gcv/validator.go]: turned InitOptions into an internal struct to hold CF functional opts and added another functional opt type to configure initOptions. lmk what you think.

File: pkg/gcv/validator.go [hunk 2]
```diff
@@ -69,6 +69,21 @@ type Validator struct {
...
 }
 
+// Stores functional options for CF client
+type initOptions struct {
+	driverArgs  []local.Arg
+	backendArgs []cfclient.BackendOpt
+	clientArgs  []cfclient.Opt
+}
+
+type InitArg = func(*initOptions)
```
Comment (briantkennedy) [inline pkg/gcv/validator.go:79]: s/InitArg/Option

File: pkg/gcv/validator.go [hunk 3]
```diff
@@ -86,14 +101,28 @@ func NewValidatorConfig(policyPaths []string, policyLibraryPath string) (*config
 func newCFClient(
 	targetHandler cfclient.TargetHandler,
 	templates []*cftemplates.ConstraintTemplate,
-	constraints []*unstructured.Unstructured) (
+	constraints []*unstructured.Unstructured,
+	args ...InitArg) (
```
Comment (briantkennedy) [inline pkg/gcv/validator.go:105]: update local var name with InitArg rename

File: pkg/gcv/validator.go (hunk 2 above)
Comment (hkundag) [reply pkg/gcv/validator.go]: done

File: pkg/gcv/validator.go (hunk 3 above)
Comment (hkundag) [reply pkg/gcv/validator.go]: done

File: pkg/gcv/validator.go [hunk 4]
```diff
@@ -86,14 +101,28 @@ func NewValidatorConfig(policyPaths []string, policyLibraryPath string) (*config
...
+	args ...Option) (
 	*cfclient.Client, error) {
-	driver := local.New(local.Tracing(false))
-	backend, err := cfclient.NewBackend(cfclient.Driver(driver))
+	options := &initOptions{}
+	for _, arg := range args {
+		arg(options)
+	}
+
+	defaultDriverArgs := []local.Arg{local.Tracing(false)}
```
Comment (briantkennedy) [inline pkg/gcv/validator.go:112]: You can set up the initOptions with this literal so you don't need the append.   Do this for all three.

File: pkg/gcv/validator.go [hunk 5]
```diff
@@ -86,14 +101,28 @@ func NewValidatorConfig(policyPaths []string, policyLibraryPath string) (*config
 func newCFClient(
 	targetHandler cfclient.TargetHandler,
 	templates []*cftemplates.ConstraintTemplate,
-	constraints []*unstructured.Unstructured) (
+	constraints []*unstructured.Unstructured,
+	args ...Option) (
```
Comment (briantkennedy) [inline pkg/gcv/validator.go:105]: s/args/opts/ or something more suited to Options

File: pkg/gcv/validator.go [hunk 6]
```diff
@@ -121,13 +150,13 @@ func newCFClient(
 }
 
 // NewValidatorFromConfig creates the validator from a config.
-func NewValidatorFromConfig(config *configs.Configuration) (*Validator, error) {
-	gcpCFClient, err := newCFClient(gcptarget.New(), config.GCPTemplates, config.GCPConstraints)
+func NewValidatorFromConfig(config *configs.Configuration, args ...Option) (*Validator, error) {
```
Comment (briantkennedy) [inline pkg/gcv/validator.go:153]: s/args/opts/ or something more suited to Options

**Review by briantkennedy — APPROVED** [review]

**Pull request merged**